File: MiscApps/NotesApp/app/src/main/java/com/example/notesapp/MainActivity.java
package com.example.notesapp;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;

import java.util.ArrayList;


public class MainActivity extends AppCompatActivity{

    private ArrayList<Note> arr = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final EditText etInput = findViewById(R.id.etInput);
        Button btnInput = findViewById(R.id.btnInput);
        ListView lv = findViewById(R.id.listView);

        final NotesAdapter adapter = new NotesAdapter(arr, this);

        lv.setAdapter(adapter);

        View.OnClickListener ocl = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                arr.add(new Note(etInput.getText().toString()));
                etInput.setText("");
                adapter.notifyDataSetChanged();
            }
        };

        btnInput.setOnClickListener(ocl);

    }
}

File: Lesson12/SaveInstanceState/app/src/main/java/com/example/saveinstancestate/MainActivity.kt
package com.example.saveinstancestate

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var count = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnInc.setOnClickListener{
            tvCount.text = (++count).toString()
        }
    }
}
